### PR TITLE
Update Upcgen recipe

### PR DESCRIPTION
**Pull request by nburmaso**
A small change to update recipe to the latest Upcgen version

@pbuehler, please take a look

Cheers,
Nazar

### CODE REVIEW

**Review by pbuehler — APPROVED** [review]

**Comment by pbuehler** [comment]
Hi @ktf, can you please approve this PR?

**Review by ktf — APPROVED** [review]

**Comment by ktf** [comment]
@pbuehler Can you please add the package to the CODEOWNERS file, adding yourself as responsible for it?

**Comment by pbuehler** [comment]
Hi @nburmaso please see previous comment by @ktf. Can you please take care?

**Comment by nburmaso** [comment]
> Hi @nburmaso please see previous comment by @ktf. Can you please take care?

Hi @pbuehler, @ktf,

I have added the line

```
upcgen* @nburmaso @pbuehler
```

but me and Paul have no write access for alidist repository -- is it ok?

**Comment by ktf** [comment]
I guess it's @alisw/pagmc?

**Comment by nburmaso** [comment]
> I guess it's @alisw/pagmc?

I have changed owners to @alisw/pagmc and requested membership

File: upcgen.sh
Comment (ktf) [inline upcgen.sh:27]: Indeed, you should add double quotes around the various `$INSTALLROOT/something`.

**Review by ktf — CHANGES_REQUESTED** [review]

File: upcgen.sh
Comment (ktf) [inline upcgen.sh:30]: While you are at it, please use `alibuild-generate-module --lib --bin` as done elsewhere.

**Comment by nburmaso** [comment]
Hi @ktf, @pbuehler,

I have fixed quotes and module generation, should be ok now.

Cheers,
Nazar

File: upcgen.sh
Comment (ktf) [inline upcgen.sh:42]:
```suggestion
```

**Review by ktf — APPROVED** [review]

File: upcgen.sh
Comment (ktf) [reply upcgen.sh]: No need for the custom PATH / LD_LIBRARY_PATH. I removed it.

File: upcgen.sh
Comment (nburmaso) [reply upcgen.sh]: Thanks a lot!

**Comment by jackal1-66** [comment]
Hi @nburmaso, I see that you have two new tags in the alice fork of upcgen which are not present in the original repository. Did you simply tweak something in the older version to make it compile in our framework? I ask you because in that case there is a guideline on how to choose the tag name: if the new tag is an offspring of [tag-24-06-22](https://github.com/nburmaso/upcgen/releases/tag/tag-24-06-22) only for ALICE use, then it should be called tag-24-06-22-alice1. Let me know if this is the case. If so I would ask you to make the modification before merging. The CI tests are not so useful in this PR because upcgen is not a requirement of O2 or AliGenenerators so it won't be built.

**Comment by nburmaso** [comment]
> Hi @nburmaso, I see that you have two new tags in the alice fork of upcgen which are not present in the original repository. Did you simply tweak something in the older version to make it compile in our framework? I ask you because in that case there is a guideline on how to choose the tag name: if the new tag is an offspring of [tag-24-06-22](https://github.com/nburmaso/upcgen/releases/tag/tag-24-06-22) only for ALICE use, then it should be called tag-24-06-22-alice1. Let me know if this is the case. If so I would ask you to make the modification before merging. The CI tests are not so useful in this PR because upcgen is not a requirement of O2 or AliGenenerators so it won't be built.

Hi @jackal1-66,

The tag isn't for ALICE-only use. This is my mistake, I should have created the tag in the original repository -- should I do so now or can we continue like this?

Cheers,
Nazar

**Comment by jackal1-66** [comment]
We can continue this way, just sync the tags on the original repo.

**Comment by nburmaso** [comment]
> We can continue this way, just sync the tags on the original repo.

Hi @jackal1-66,

Done :) Could we please merge this PR if it is ok? Also, could you please add me to @alisw/pagmc?

Cheers,
Nazar

**Review by sawenzel — APPROVED** [review]

**Pull request merged**
